lib/mp4/transmuxer.js

```diff
@@ -763,7 +788,27 @@ CoalesceStream.prototype.flush = function(flushSource) {
   this.pendingMetadata.length = 0;
 
   // Emit the built segment
+  // We include captions and ID3 tags for backwards compatibility,
+  // ideally we should send only video and audio in the data event
   this.trigger('data', event);
+  // Emit each caption to the outside world
+  // Ideally, this would happen immediately on parsing captions,
+  // but we need to ensure that video data is sent back first
+  // so that caption timing can be adjusted to match video timing
+  for (i = 0; i < event.captions.length; i++) {
```

Comment:
Why do we need to send them out as individual events instead of a single event with just all the captions?  Same question for ID3 tags

Reply:
This matches the behavior of the partialTransmuxer which will get one caption/ID3 as they are encountered